File: DerfAccessibilityService.java
package org.derf.messenger;

import android.accessibilityservice.AccessibilityService;
import android.view.accessibility.AccessibilityEvent;
import com.chaquo.python.Python;
import com.chaquo.python.PyObject;
import java.util.List;

public class DerfAccessibilityService extends AccessibilityService {
    @Override
    public void onAccessibilityEvent(AccessibilityEvent event) {
        if (event.getEventType() == AccessibilityEvent.TYPE_VIEW_TEXT_CHANGED ||
            event.getEventType() == AccessibilityEvent.TYPE_VIEW_TEXT_SELECTION_CHANGED) {

            List<CharSequence> texts = event.getText();
            if (texts != null) {
                for (CharSequence text : texts) {
                    if (text != null && text.toString().contains("DERF:V1:")) {
                        try {
                            if (Python.isStarted()) {
                                PyObject module = Python.getInstance().getModule("Derf");
                                module.callAttr("safe_copy", text.toString());
                            }
                        } catch (Exception e) {
                            e.printStackTrace();
                        }
                        break;
                    }
                }
            }
        }
    }

    @Override
    public void onInterrupt() {}
}

File: android/app/src/main/java/com/derf/pq/DerfAccessibilityService.kt
package com.derf.pq

import android.accessibilityservice.AccessibilityService
import android.view.accessibility.AccessibilityEvent
import com.chaquo.python.Python

class DerfAccessibilityService : AccessibilityService() {
    override fun onAccessibilityEvent(event: AccessibilityEvent?) {
        if (event == null) return
        if (event.eventType == AccessibilityEvent.TYPE_VIEW_TEXT_CHANGED ||
            event.eventType == AccessibilityEvent.TYPE_VIEW_TEXT_SELECTION_CHANGED) {

            val texts = event.text
            if (texts != null) {
                for (text in texts) {
                    if (text != null && text.toString().contains("DERF:V1:")) {
                        try {
                            if (Python.isStarted()) {
                                val module = Python.getInstance().getModule("Derf")
                                module.callAttr("safe_copy", text.toString())
                            }
                        } catch (e: Exception) {
                            e.printStackTrace()
                        }
                        break
                    }
                }
            }
        }
    }

    override fun onInterrupt() {}
}
